aws/signer/v4/middleware.go

```diff
@@ -247,18 +247,18 @@ func (s *SignHTTPRequestMiddleware) HandleFinalize(ctx context.Context, in middl
 		return out, metadata, &SigningError{Err: fmt.Errorf("failed to retrieve credentials: %w", err)}
 	}
 
-	err = s.signer.SignHTTP(ctx, credentials, req.Request, payloadHash, signingName, signingRegion, sdk.NowTime(), s.addSignerOptions)
+	err = s.signer.SignHTTP(ctx, credentials, req.Request, payloadHash, signingName, signingRegion, sdk.NowTime(),
+		func(o *SignerOptions) {
+			o.Logger = middleware.GetLogger(ctx)
+			o.LogSigning = s.logSigning
+		})
```

Comment:
If you wanted to avoid the function enclosure, we could switch HandleFinalize to be a value receiver. Then you could override the logger on that copy of the middleware with the context value.

Reply:
Sounds good can update to use that pattern.

Reply:
hmm actually this would require adding `Logger` member to the middleware type, which is probably out of place since its not a value that would only be present for this method call.  Might be better to just keep the closure at that point.